backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/process/ModCntr_Specific_Price_Update_Selection.java

```diff
@@ -132,31 +133,35 @@ public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
 
 	private void updatePrice(final ModCntrSpecificPriceId contractPriceId)
 	{
-		final ModCntrSpecificPrice newContractPrice;
-
-		if (p_asNewPrice)
-		{
-			newContractPrice = modularContractPriceService.cloneById(contractPriceId, contractPrice -> contractPrice.toBuilder()
-					.amount(Money.of(p_price, p_C_Currency_ID))
-					.uomId(p_C_UOM_ID)
-					.minValue(p_minValue)
-					.build());
-		}
-		else
-		{
-			newContractPrice = modularContractPriceService.updateById(contractPriceId, contractPrice -> contractPrice.toBuilder()
-					.amount(Money.of(p_price, p_C_Currency_ID))
-					.uomId(p_C_UOM_ID)
-					.minValue(p_minValue)
-					.build());
-		}
-
-		contractLogService.updatePriceAndAmount(ModCntrLogPriceUpdateRequest.builder()
-						.unitPrice(newContractPrice.getProductPrice())
-						.flatrateTermId(newContractPrice.flatrateTermId())
-						.modularContractModuleId(newContractPrice.modularContractModuleId())
-						.build(),
-				logHandlerRegistry);
+		trxManager.assertThreadInheritedTrxNotExists();
+		trxManager.runInThreadInheritedTrx(() -> {
```

Comment:
because now we run in its own trx, 
wondering if we should wrap this by try/catch and send a notification in case of any error.

Reply:
Usually all contracts should be updated the same way, so if it fails it should already fail on the first contract.